neurom/core/_neuron.py

```diff
@@ -28,16 +28,176 @@
 
 """Neuron classes and functions."""
 
-from copy import deepcopy
+from collections import deque
 from itertools import chain
 
+import morphio
 import numpy as np
 from neurom import morphmath
-from neurom.core._soma import Soma
+from neurom.core._soma import make_soma
 from neurom.core.dataformat import COLS
-from neurom.utils import memoize
+from neurom.core.types import NeuriteIter, NeuriteType
+
+
+class Section:
+    """Simple recursive tree class."""
+
+    def __init__(self, morphio_section):
+        """The section constructor."""
+        self.morphio_section = morphio_section
+
+    @property
+    def id(self):
+        """Returns the section ID."""
+        return self.morphio_section.id
+
+    @property
+    def parent(self):
+        """Returns the parent section if non root section else None."""
+        return None if self.morphio_section.is_root else Section(self.morphio_section.parent)
+
+    @property
+    def children(self):
+        """Returns a list of child section."""
+        return [Section(child) for child in self.morphio_section.children]
+
+    def append_section(self, section):
+        """Appends a section.
+
+        Args:
+            section (morphio.Section|morphio.mut.Section|Section|morphio.PointLevel): a section
+        """
+        if isinstance(section, Section):
+            return self.morphio_section.append_section(section.morphio_section)
+        return self.morphio_section.append_section(section)
+
+    def is_forking_point(self):
+        """Is tree a forking point?"""
+        return len(self.children) > 1
+
+    def is_bifurcation_point(self):
```

Comment:
can't we have more than 2 children ?

Reply:
We can but this comes from NeuroM v1. `Tree` class has been deleted and its functionality was copy-pasted to `Section`. From what I saw/heard this method is very important in NeuroM v1.